README.md

```diff
@@ -1,26 +1,85 @@
-# <img src=	https://upload.wikimedia.org/wikipedia/commons/7/7a/ROS_cat.png width="250" height="140">
+<div class="row">
+  <div class="column">
 
+  </div>
+  <div class="column">
+
+  </div>
+</div>
+<p float="left">
+    <a href="https://revolve.no/"><img align=left src="https://raw.githubusercontent.com/RevolveNTNU/RosNet/86-improve-readme/.github/main/revolve.svg" width="30%" height="100"/></a>
+    <a href="https://ros.org/"><img align=right margin src="https://raw.githubusercontent.com/RevolveNTNU/RosNet/86-improve-readme/.github/main/ros.png" height="100"/></a>
+</p>
+<br>
+<br>
+<br>
+<br>
+<br>
+
+[![Build, run tests and generate report](https://github.com/RevolveNTNU/RosNet/actions/workflows/build_and_test.yml/badge.svg)](https://github.com/RevolveNTNU/RosNet/actions/workflows/build_and_test.yml)
+[![Coverage Status](https://coveralls.io/repos/github/RevolveNTNU/RosNet/badge.svg?branch=master)](https://coveralls.io/github/RevolveNTNU/RosNet?branch=master)
 # RosNet
 ## What is RosNet?
-RosNet is a .NET library used for parsing RosBags to a C# object. We are thinking of adding support for writing as well. 
+RosNet is a .NET library for deserializing rosbags version 2.0 to C#-objects.
 
 ## How to use the library:
-<!-- How to include library in project-->
+For full API documentation go to: https://revolventnu.github.io/RosNet/
+<!-- User documentation-->
+To use RosNet in a C#-based application: Install RosNet as a NuGet Package
+
+### Using statement
 ```C#
 using RosNet.DataModel;
-
-# Convert a ROSBag to a C# object
+```
+### Convert a ROSBag to a C# object
+```C#
 var rosBag = new RosBag(path);
 rosBag.Read();
-
-# Obtain possible ROS topics and fields in the topics
+```
+### Obtain possible ROS topics and fields in the topics
+```C#
 Dictionary<string, List<string>> fieldNamesByTopic = rosBag.GetConnectionFields();
+```
 
-# Obtain list of tuples with timestamp and fieldvalue
+### Obtain list of tuples with timestamp and fieldvalue
+```C#
 List<(Time, FieldValue)> timeSeries = rosBag.GetTimeSeries(topic, fieldName);
 ```
-For full API documentation go to: https://revolventnu.github.io/RosNet/
 
+### FieldValue
+
+FieldValue is a custom data type with a 
+
+- string name -  name of the field
+- PrimitiveType DataType -  datatype of the value. PrimitiveType is an Enum with possible values: Bool, Int8, Uint8, Int16, Unit16, Int32, Uint32, Int64, Uint64, Float32, Float64, String, Time, Duration, Byte, Char and Array. The datatypes correspond to the [standard datatypes](https://wiki.ros.org/msg) in ROS messages and Array used for arrays of values.
+- byte[] Value - is the value of the field
```

Comment:
```suggestion
You will primarily interact with `FieldValue` which has the following fields

- `string Name` -  name of the field
- `PrimitiveType DataType` -  datatype of the value. PrimitiveType is an Enum with possible values: Bool, Int8, Uint8, Int16, Unit16, Int32, Uint32, Int64, Uint64, Float32, Float64, String, Time, Duration, Byte, Char and Array. The datatypes correspond to the [standard datatypes](https://wiki.ros.org/msg) in ROS messages and Array used for arrays of values.
- `byte[] Value` - is the value of the field
```

Calling it a "custom data type" is not necessary, and could rather be a tiny bit confusing with current development of e.g. data classes in Python, (usually called records in C# lingo though)